openlibrary/core/models.py

```diff
@@ -409,6 +409,8 @@ def from_isbn(cls, isbn):
             return None  # consider raising ValueError
 
         isbn13 = to_isbn_13(isbn)
+        if isbn13 is None:
```

Comment:
Could a valid isbn-10 cause this function to return None?

Reply:
I tested it with a valid ISBN-10 in my local environment (http://localhost:8080/search?debug=true&isbn=0863224334 - in the sample data for the development version). It works fine.

The to_isbn_13 function normalises a 10-digit ISBN and returns str|None. It uses the normalize_isbn function in utils/isbn.py which also has a return type of str|None, and in turn uses canonical from isbnlib/_core.py.

Basically if it is a valid ISBN-10 it returns a string value, if it is a valid ISBN-13 it returns a string value. If it _isn't_ a valid ISBN10 it will return None.

I haven't tested this exhaustively with lots of differently formatted ISBN-10 strings but the to_isbn_13 function proxies to normalize_isbn and ultimately to isbnlib's canonical function. to_isbn_13 also calls utils/isbn.py isbn_10_to_isbn_13 to reformat the ISBN-10 to ISBN-13. That function uses validation checks to validate the ISBN-10 and can return None but only for _invalid_ ISBN-10.

I believe that covers the various cases and answers your question.

Reply:
@agmckee thanks for the PR, I did a rushed version of basically the same fix and pushed it before I noticed yours.

Thanks for the extra testing you did, I agree with everything you describe above about the existing methods from my static review of the methods, and testing `isbnlib.canonical()`  👍🏽